### PR TITLE
Support external graphql plugins

### PR DESCRIPTION
**Pull request by sdobz**
Fixes issue #3 and issue #7 by replicating  in the context of this plugin

A key discovery was this issue:
https://github.com/gatsbyjs/gatsby/issues/5663

Which pointed me to the file
https://github.com/gatsbyjs/gatsby/blob/2.13.30/packages/gatsby/src/query/query-compiler.js

By duplicating that code into this plugin we have mostly the same file resolution and should discover any fragments that gatsby can

### CODE REVIEW

**Review by xndyz — APPROVED** [review]

**Comment by xndyz** [comment]
@cometkim this PR should solve the breaking issue where anyone using fragments would not be able to use the plugin at all. I think it's ready to merge 👍

File: package.json [hunk 1]
```diff
@@ -25,15 +25,19 @@
...
   "peerDependencies": {
     "gatsby": ">=2.0.0",
+    "gatsby-cli": ">=2.0.0",
     "graphql": "^0.13.0 || ^14.0.0"
   },
   "dependencies": {
     "@graphql-codegen/core": "^1.7.0",
     "@graphql-codegen/typescript": "^1.7.0",
     "@graphql-codegen/typescript-operations": "^1.7.0",
+    "@types/normalize-path": "^3.0.0",
```
Comment (cometkim) [inline package.json]: Can move this into devDeps?

File: gatsby-node.ts [hunk 2]
```diff
@@ -10,6 +10,13 @@ import { loadDocuments, loadSchema } from 'graphql-toolkit';
 import { GatsbyNode } from 'gatsby';
 // @ts-ignore
 import { graphql, introspectionQuery } from 'gatsby/graphql';
+// @ts-ignore
+import getGatsbyDependents from "gatsby/dist/utils/gatsby-dependents.js"
+// @ts-ignore
+import report from "gatsby-cli/lib/reporter"
```
Comment (cometkim) [inline gatsby-node.ts]: If I understood this correctly, the reporter already included in [Node API helpers](https://www.gatsbyjs.org/docs/node-api-helpers/#reporter), so we don't need to import gatsby-cli for it.

File: gatsby-node.ts [hunk 3]
```diff
@@ -10,6 +10,13 @@ import { loadDocuments, loadSchema } from 'graphql-toolkit';
 import { GatsbyNode } from 'gatsby';
 // @ts-ignore
 import { graphql, introspectionQuery } from 'gatsby/graphql';
+// @ts-ignore
+import getGatsbyDependents from "gatsby/dist/utils/gatsby-dependents.js"
```
Comment (cometkim) [inline gatsby-node.ts]:
```suggestion
import getGatsbyDependents from 'gatsby/dist/utils/gatsby-dependents'
```

**Review by cometkim — COMMENTED** [review]
Thanks @sdobz, for this nice PR! I will carefully merge this with #6.

File: gatsby-node.ts (hunk 2 above)
Comment (sdobz) [reply gatsby-node.ts]: Indeed! I have rewritten the code to use the included reporter.

File: package.json (hunk 1 above)
Comment (sdobz) [reply package.json]: Done, good catch

File: gatsby-node.ts (hunk 3 above)
Comment (sdobz) [reply gatsby-node.ts]: Done, the other " have been turned to '

**Comment by cometkim** [comment]
I will merge this in #11, Thank you for your contribution @sdobz!

**Pull request closed without merging**